boa/src/syntax/parser/expression/unary.rs

```diff
@@ -61,11 +61,23 @@ where
     fn parse(self, cursor: &mut Cursor<R>) -> ParseResult {
         let _timer = BoaProfiler::global().start_event("UnaryExpression", "Parsing");
 
-        let tok = cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?;
+        // TODO: can we avoid cloning?
+        let tok = cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?.clone();
```

Comment:
why do we have to clone here?

Reply:
Because we later use the token() span as part of the error message, I've changed this to an alternative mechanism which avoids the clone (still needs an implicit copy).

Reply:
should be fine start is a usize at worst a u64 so I don't think this is ever going to be an issue :)